code/modules/mapping/mapping_helpers.dm

```diff
@@ -235,26 +235,39 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
 	name = "Dead Body placer"
 	late = TRUE
 	icon_state = "deadbodyplacer"
-	var/bodycount = 2 //number of bodies to spawn
+	/// number of bodies to spawn 
+	var/bodycount = 2
+	/// set it TRUE in mapping if you want to put it into a specific container (when you don't want a random morgue)
+	var/no_random = FALSE
 
 /obj/effect/mapping_helpers/dead_body_placer/LateInitialize()
 	var/area/a = get_area(src)
 	var/list/trays = list()
-	for (var/i in a.contents)
-		if (istype(i, /obj/structure/bodycontainer/morgue))
-			trays += i
-	if(!trays.len)
-		log_mapping("[src] at [x],[y] could not find any morgues.")
-		return
-	for (var/i = 1 to bodycount)
-		var/obj/structure/bodycontainer/morgue/j = pick(trays)
-		var/mob/living/carbon/human/h = new /mob/living/carbon/human(j, 1)
-		h.death()
-		for (var/part in h.internal_organs) //randomly remove organs from each body, set those we keep to be in stasis
-			if (prob(40))
-				qdel(part)
-			else
-				var/obj/item/organ/O = part
-				O.organ_flags |= ORGAN_FROZEN
-		j.update_icon()
+	if(!no_random)
+		for (var/obj/structure/bodycontainer/morgue/morgue in a.contents)
```

Comment:
Don't loop through area contents, this is incredibly slow. I know it did it before, but it needs a better method than this.

Reply:
A view() check or even only placing in adjacent would be better

Reply:
the change request you did actually needs to include map change. I don't want to do that right now...

Reply:
actually, it's a bad idea when it's in maint. I will change the maint map thing at least.